plasma_framework/python_tests/tests/contracts/root_chain/test_process_exits.py

```diff
@@ -1002,3 +1002,49 @@ def test_challenged_standard_exit_does_not_block_ife_output_exit(testlang, plasm
     # and does not get the tokens back
     alice_token_balance = token.balanceOf(alice.address)
     assert alice_token_balance == alice_token_balance_before
+
+
+def test_after_canonical_ife_is_finalized_inputs_are_not_exited_when_ife_is_restarted_and_non_canonical(testlang, plasma_framework, token):
+    alice, bob, amount = testlang.accounts[0], testlang.accounts[1], 100
+    alice_deposit_id = testlang.deposit(alice, amount)
+    bob_deposit_id = testlang.deposit(bob, amount)
+
+    spend_id = testlang.spend_utxo([alice_deposit_id, bob_deposit_id], [alice, bob], [(alice.address, NULL_ADDRESS, amount * 2)])
+
+    alice_eth_balance_before = testlang.get_balance(alice)
+
+    testlang.start_in_flight_exit(spend_id)
+    testlang.piggyback_in_flight_exit_output(spend_id, 0, alice)
+    testlang.forward_timestamp(2 * MIN_EXIT_PERIOD + 1)
+    testlang.process_exits(NULL_ADDRESS, 0, 1)
+
+    # alice exits with her Eth output
+    alice_eth_balance = testlang.get_balance(alice)
+    assert alice_eth_balance == alice_eth_balance_before + amount * 2
```

Comment:
might worth to put a var on `amount * 2` to clarify what that is (use that on the output when `spend_utxo` too)